nomos-services/storage/src/backends/rocksdb.rs

```diff
@@ -0,0 +1,255 @@
+// std
+use std::path::PathBuf;
+use std::{marker::PhantomData, sync::Arc};
+// crates
+use async_trait::async_trait;
+use bytes::Bytes;
+pub use rocksdb::Error;
+use rocksdb::{Options, DB};
+// internal
+use super::{StorageBackend, StorageSerde, StorageTransaction};
+
+/// Rocks backend setting
+#[derive(Clone, Debug)]
+pub struct RocksBackendSettings {
+    /// File path to the db file
+    pub db_path: PathBuf,
+    pub read_only: bool,
+    pub column_family: Option<String>,
+}
+
+/// Rocks transaction type
+// Do not use `TransactionDB` here, because rocksdb's `TransactionDB` does not support open by read-only mode.
+// Thus, we cannot open the same db in two or more processes.
```

Comment:
So, we cannot have a process writing and a process reading?

Reply:
We can because we use `DB` here, not the `TransactionDB`.